libs/backend/src/cast_vote_records/export.ts

```diff
@@ -668,5 +679,81 @@ export async function exportCastVoteRecordsToUsbDrive(
     return exportSignatureFileResult;
   }
 
+  clearDoesUsbDriveRequireCastVoteRecordSyncCachedResult();
+
   return ok();
 }
+
+/**
+ * Returns whether a USB drive is inserted and requires a cast vote record sync because the cast
+ * vote records on it don't match the cast vote records on the machine. Only relevant for machines
+ * that continuously export cast vote records.
+ *
+ * Because this function 1) requires reading data from the USB drive and 2) is polled by consumers,
+ * we use a caching mechanism to avoid constantly reading from the USB drive. We recompute whenever
+ * 1) the USB drive status changes (e.g. on insertion and removal) or 2) the cache is explicitly
+ * cleared (e.g. after cast vote record export).
+ */
+export async function doesUsbDriveRequireCastVoteRecordSync(
+  scannerStore: ScannerStore & {
+    getBallotsCounted: () => number;
+    getExportDirectoryName: NonNullable<ScannerStore['getExportDirectoryName']>;
+  },
+  usbDriveStatus: UsbDriveStatus
+): Promise<boolean> {
+  if (
+    doesUsbDriveRequireCastVoteRecordSyncCachedResult &&
+    doesUsbDriveRequireCastVoteRecordSyncCachedResult.previousUsbDriveStatus
+      .status === usbDriveStatus.status
+  ) {
+    return doesUsbDriveRequireCastVoteRecordSyncCachedResult.previousValue;
+  }
+
+  const value = await (async () => {
+    if (usbDriveStatus.status !== 'mounted') {
+      return false;
+    }
+    const usbMountPoint = usbDriveStatus.mountPoint;
+
+    const electionDefinition = scannerStore.getElectionDefinition();
+    if (!electionDefinition) {
+      return false;
+    }
+    const exportDirectoryName = scannerStore.getExportDirectoryName();
+    if (!exportDirectoryName) {
+      return false;
+    }
+    const ballotsCounted = scannerStore.getBallotsCounted();
+    if (ballotsCounted === 0) {
+      return false;
+    }
+    const castVoteRecordRootHash = scannerStore.getCastVoteRecordRootHash();
+
+    const { election, electionHash } = electionDefinition;
+    const exportDirectoryPath = path.join(
+      usbMountPoint,
+      SCANNER_RESULTS_FOLDER,
+      generateElectionBasedSubfolderName(election, electionHash),
+      exportDirectoryName
+    );
+    const metadataResult =
+      await readCastVoteRecordExportMetadata(exportDirectoryPath);
+    if (metadataResult.isErr()) {
+      return true;
+    }
+    const castVoteRecordExportMetadata = metadataResult.ok();
+    if (
+      castVoteRecordExportMetadata.castVoteRecordRootHash !==
+      castVoteRecordRootHash
+    ) {
+      return true;
+    }
+    return false;
+  })();
+  doesUsbDriveRequireCastVoteRecordSyncCachedResult = {
+    previousUsbDriveStatus: usbDriveStatus,
+    previousValue: value,
+  };
+
+  return value;
+}
```

Comment:
Going to add more complete tests for this function when I add `libs/backend` tests in a follow-up PR. But I do add some basic tests in this PR after wiring everything together.